sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/AzureEventHubCommonProperties.java

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.autoconfigure.eventhubs;
+
+import com.azure.core.amqp.implementation.ConnectionStringProperties;
+import com.azure.spring.cloud.autoconfigure.properties.AbstractAzureAmqpConfigurationProperties;
+
+/**
+ * Azure Event Hub related properties.
+ */
+public abstract class AzureEventHubCommonProperties extends AbstractAzureAmqpConfigurationProperties {
+
+    protected String domainName = "servicebus.windows.net";
+    protected String namespace;
+    protected String eventHubName;
+    protected String connectionString;
+    protected String customEndpointAddress;
+    protected Integer prefetchCount;
+
+    protected String extractFQDNFromConnectionString() {
+        if (this.connectionString == null) {
+            return null;
+        }
+        return new ConnectionStringProperties(this.connectionString).getEndpoint().getHost();
```

Comment:
take a look of `com.azure.resourcemanager.appservice.models.ConnectionString` and `com.azure.resourcemanager.appservice.models.ConnectionStringType`

I'm thinking maybe we can have a ConnectionString class to represent all the connection strings 


```
ConnectionStringType{
EVENT_HUB(" Endpoint=sb://<FQDN>/;SharedAccessKeyName=<KeyName>;SharedAccessKey=<KeyValue>"),
SERVICE_BUS("")
}

class ConnectionString{
public ConnectionString(String str, ConnectionStringType type){} // with the schema defined by each type, this class knows how to parse the string
public String getSegment(String key){}
}
```

Reply:
Let's discuss this.

Reply:
refactored.